web/src/features/AppConfig/components/AppConfig.tsx

```diff
@@ -670,26 +720,26 @@ class AppConfig extends Component {
                           appSlug={this.props?.app?.slug}
                           chartPath={this.props?.app?.chartPath || ""}
                           downloadClicked={download}
-                          downloadError={downloadError}
-                          isDownloading={isDownloading}
                           hideHelmDeployModal={() => {
                             this.setState({ showHelmDeployModal: false });
                             clearDownloadError();
                           }}
                           registryUsername={
-                            this.props?.app?.credentials?.username
+                            this.props?.app?.credentials?.username || ""
```

Comment:
Could probably remove the optional chaining here but I think we should continue typing App out some more and then figure out a way to initialize it with sane defaults.